core/src/test/java/org/opensearch/sql/common/utils/StringUtilsTest.java

```diff
@@ -0,0 +1,44 @@
+package org.opensearch.sql.common.utils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.opensearch.sql.common.utils.StringUtils.unquoteText;
+
+import org.junit.jupiter.api.Test;
+
+class StringUtilsTest {
+  @Test
+  void unquoteTest() {
+    assertEquals("test\\'s", unquoteText("'test\\\\'s'"));
+    assertEquals("test\\'s", unquoteText("'test\\\\\\'s'"));
+    assertEquals("test\\\\'s", unquoteText("'test\\\\\\\\'s'"));
+    assertEquals("test\\\\'s", unquoteText("'test\\\\\\\\\\'s'"));
+
+    assertEquals("test\\'s", unquoteText("`test\\'s`"));
+    assertEquals("test\\'s", unquoteText("`test\\\\'s`"));
+    assertEquals("test\\\\'s", unquoteText("`test\\\\\\'s`"));
+    assertEquals("test\\\\'s", unquoteText("`test\\\\\\\\'s`"));
+
+    assertEquals("test\\'s", unquoteText("\"test\\'s\""));
+    assertEquals("test\\'s", unquoteText("\"test\\\\'s\""));
+    assertEquals("test\\\\'s", unquoteText("\"test\\\\\\'s\""));
+    assertEquals("test\\\\'s", unquoteText("\"test\\\\\\\\'s\""));
+
+    assertEquals("'test'", unquoteText("\"'test'\""));
+    assertEquals("\"test\"", unquoteText("'\"test\"'"));
+    assertEquals("test", unquoteText("'test'"));
+    assertEquals("test", unquoteText("\"test\""));
+
+    assertEquals("test's", unquoteText("'test''s'"));
+    assertEquals("test''s", unquoteText("'test''''s'"));
+    assertEquals("test\"s", unquoteText("\"test\"\"s\""));
+    assertEquals("test\"\"s", unquoteText("\"test\"\"\"\"s\""));
+
+    assertEquals("test''s", unquoteText("\"test''s\""));
+    assertEquals("test''''s", unquoteText("\"test''''s\""));
+    assertEquals("test\"\"s", unquoteText("'test\"\"s'"));
+    assertEquals("test\"\"\"\"s", unquoteText("'test\"\"\"\"s'"));
+
+    // Tests case of unquoted string sent to unquoteText function
+    assertEquals("test", unquoteText("test"));
+  }
```

Comment:
More tests for god of tests!
Tests with backtick `` ` ``, tests with combined `'` and `"`, mixes of `'`/`"`/`\` and `'`/`"`/`\` in the beginning and in the end of line wrapped by `'`/`"`/`` ` ``/none and so on.